### DISCUSSION TITLE
Option picker navigate to screen after selection

### DISCUSSION THREAD

**Discussion by masterfixedit** [Budibase/budibase#13745] [Help] [answered]
Hi,

I'm trying to build a smartphone repair form. I use an Option Picker to select the brand, when the brand is selected it must navigate to another page where the correct model can be selected.
I tried it with actions but it doesn't work the way i want. If i choose the last brand in the option picker it navigates to all the pages of the other brands first and as last the page i want. Then i need to close all the other pages (in modal) too. Ultimately i want to mail the choices made to the customer and myself.

**Comment by mikesealey** [comment, marked as answer]
Hi @masterfixedit I'm sorry to hear you're having trouble. 

Your issue sounds similar to https://github.com/Budibase/budibase/discussions/13749 that I've just written up for another user. In their case their user will pick a State, and then a Province. It sounds like your users will pick a brand and then a model dependant on brand.

If you're writing your own tables in budibase it will be even easier because you wont need to call APIs, just your own tables. You can just a Navigate To action that takes the value from the form field, and is triggered On Change.
